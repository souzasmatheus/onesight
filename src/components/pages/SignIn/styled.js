import styled from 'styled-components';
import CardActions from '@material-ui/core/CardActions';
import CardHeader from '@material-ui/core/CardHeader';

export const StyledCardActions = styled(CardActions)`
  justify-content: flex-end;
`;

export const StyledCardHeader = styled(CardHeader)`
  &.MuiCardHeader-root {
    padding: 16px 16px 0;
  }
`;
