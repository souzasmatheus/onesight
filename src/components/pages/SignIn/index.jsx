import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';

import CardContent from '@material-ui/core/CardContent';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Alert from '@material-ui/lab/Alert';
import CircularProgress from '@material-ui/core/CircularProgress';

import { isValidEmail, isValidPassword } from '~/lib/validation';

import { auth } from '~/services/firebase';

import { setUser } from '~/feature/user/UserSlice';

import { StyledCardActions, StyledCardHeader } from './styled';

import { StyledLink } from '~/components/atoms';
import { Form } from '~/components/molecules';

const SignIn = () => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [isValid, setValid] = useState(false);
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (isValidEmail(email) && isValidPassword(password)) {
      setValid(true);
    } else {
      setValid(false);
    }
  }, [email, password]);

  const handleSignIn = async () => {
    setError('');
    setLoading(true);

    try {
      const { user } = await auth.signInWithEmailAndPassword(email, password);
      dispatch(setUser(user.email));
    } catch (e) {
      setLoading(false);
      setError(e.message);
    }
  };

  return (
    <Form>
      {error && (
        <Alert variant="outlined" severity="error">
          {error}
        </Alert>
      )}
      <StyledCardHeader
        title="Sign in"
        subheader={
          <p>
            Not registered yet? <StyledLink to="/sign-up">Sign up</StyledLink>!
          </p>
        }
      />
      <CardContent>
        <TextField
          margin="dense"
          type="email"
          label="E-mail"
          fullWidth
          onChange={(e) => setEmail(e.target.value)}
        />
        <TextField
          margin="dense"
          type="password"
          label="Password"
          fullWidth
          onChange={(e) => setPassword(e.target.value)}
        />
      </CardContent>
      <StyledCardActions>
        {loading ? (
          <CircularProgress />
        ) : (
          <Button disabled={!isValid} variant="outlined" onClick={handleSignIn}>
            Sign in
          </Button>
        )}
      </StyledCardActions>
    </Form>
  );
};

export default SignIn;
