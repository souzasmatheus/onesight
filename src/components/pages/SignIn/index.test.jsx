import React from 'react';
import { render } from '@testing-library/react';

import {
  MockedTheme,
  MockedRouter,
  MockedStore,
} from '~/../test/mocks/providers';

import SignIn from './index';

describe('<SignIn />', () => {
  it('should match snapshot test', () => {
    const tree = render(
      <MockedRouter>
        <MockedStore>
          <MockedTheme>
            <SignIn />
          </MockedTheme>
        </MockedStore>
      </MockedRouter>,
    );

    expect(tree).toMatchSnapshot();
  });
});
