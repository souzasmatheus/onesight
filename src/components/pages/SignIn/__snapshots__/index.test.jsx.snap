// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should match snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKsewC hJJWkx"
      >
        <div
          class="MuiPaper-root MuiCard-root sc-jSgupP boYLPn MuiPaper-elevation1 MuiPaper-rounded"
        >
          
          <div
            class="MuiCardHeader-root sc-gsTCUz iNdqRX"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Sign in
              </span>
              <span
                class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                <p>
                  Not registered yet? 
                  <a
                    class="sc-dlfnbm drfFgG"
                    href="/sign-up"
                  >
                    Sign up
                  </a>
                  !
                </p>
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
                data-shrink="false"
              >
                E-mail
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
                data-shrink="false"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root sc-bdfBwQ dWoOti MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKsewC hJJWkx"
    >
      <div
        class="MuiPaper-root MuiCard-root sc-jSgupP boYLPn MuiPaper-elevation1 MuiPaper-rounded"
      >
        
        <div
          class="MuiCardHeader-root sc-gsTCUz iNdqRX"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              Sign in
            </span>
            <span
              class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              <p>
                Not registered yet? 
                <a
                  class="sc-dlfnbm drfFgG"
                  href="/sign-up"
                >
                  Sign up
                </a>
                !
              </p>
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
              data-shrink="false"
            >
              E-mail
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
              data-shrink="false"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiCardActions-root sc-bdfBwQ dWoOti MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sign in
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
